packages/server/api/src/app/telemetry/telemetry.ts

```diff
@@ -15,42 +15,71 @@ import {
 } from './logzio-collector';
 import { TelemetryEvent } from './telemetry-event';
 
+const telemetryMode = system.get<TelemetryMode>(AppSystemProp.TELEMETRY_MODE);
 const telemetryCollectorUrl = system.get<string>(
   AppSystemProp.TELEMETRY_COLLECTOR_URL,
 );
 const logzioMetricToken = system.get<string>(
   SharedSystemProp.LOGZIO_METRICS_TOKEN,
 );
+
 const version = system.get<string>(SharedSystemProp.VERSION);
 
+export enum TelemetryMode {
+  DISABLED = 'DISABLED',
+  COLLECTOR = 'COLLECTOR',
+  LOGZIO = 'LOGZIO',
+}
+
+let isTelemetryEnabled = true;
 let environmentId: UUID | undefined;
 export const telemetry = {
   async start(getEnvironmentId: () => Promise<UUID>): Promise<void> {
-    if (!telemetryCollectorUrl && !logzioMetricToken) {
-      throw new Error(
-        'Telemetry is enabled, but neither TELEMETRY_COLLECTOR_URL nor LOGZIO_METRICS_TOKEN is defined.',
-      );
+    switch (telemetryMode) {
+      case TelemetryMode.COLLECTOR: {
+        if (!telemetryCollectorUrl) {
+          throw new Error(
+            `System property OPS_${AppSystemProp.TELEMETRY_COLLECTOR_URL} is not defined, but telemetry mode is set to ${TelemetryMode.COLLECTOR}.`,
+          );
+        }
+        break;
+      }
+      case TelemetryMode.LOGZIO: {
+        if (!logzioMetricToken) {
+          throw new Error(
+            `System property OPS_${SharedSystemProp.LOGZIO_METRICS_TOKEN} is not defined, but telemetry mode is set to ${TelemetryMode.LOGZIO}.`,
+          );
+        }
+        break;
+      }
+      default: {
+        logger.debug('Telemetry is disabled.');
+        isTelemetryEnabled = false;
+        return;
+      }
     }
 
     environmentId = await getEnvironmentId();
-    if (telemetryCollectorUrl) {
+    if (telemetryMode === TelemetryMode.COLLECTOR) {
+      logger.info('Using telemetry collector to save the telemetry events.');
       return;
     }
 
+    logger.info('Using Logz.io to save the telemetry events.');
     startMetricsCollector();
```

Comment:
You can move these inside the switch-case,  and call the getEnvironmentId before the switch-case.